src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1426,5 +1426,24 @@ private static bool NormalizeTaskTypesCore(CSharpCompilation compilation, ref Ar
             arrayType = arrayType.WithElementType(elementType);
             return true;
         }
+
+        internal static Cci.TypeReferenceWithAttributes GetTypeRefWithAttributes(
+            CSharpCompilation declaringCompilation,
+            TypeSymbol type,
+            Cci.ITypeReference typeRef)
```

Comment:
Perhaps `GetTypeRefWithAttributes(this TypeSymbol type, CSharpCompilation compilation, Cci.ITypeReference typeRef)` so callers can use `type.GetTypeRefWithAttributes(DeclaringCompilation, typeRef)`.